File: Luna.Commons/Models/CustomProperty.cs
using System.Collections.Generic;

namespace Luna.Commons.Models
{
    public class CustomProperty : ModelBase
    {
        public virtual int Order { get; set; }
    
        public virtual int CustomSectionId { get; set; }
        public virtual int TypeId { get; set; }
        public virtual int? RaceId { get; set; }
        
        public virtual CustomPropertyType Type { get; set; }
        public virtual CustomSection CustomSection { get; set; }
        public virtual Race Race { get; set; }
        
        public virtual IList<CustomPropertyHasCustomField> CustomPropertyHasCustomFields { get; set; }
    }
}

File: Luna.Web/_wwwsrc/js/Addons/Models/CustomProperty.ts
import {CustomPropertyType} from "./CustomPropertyType";
import {CustomPropertyHasCustomFields} from "./CustomPropertyHasCustomFields";

export class CustomProperty {
    public id: number;
    public name: string;
    public description: string;
    
    public order: number;
    
    public type: CustomPropertyType;

    public customPropertyHasCustomFields: CustomPropertyHasCustomFields[];

    constructor(data?) {
        if (data) {
            this.id = data.id;
            this.name = data.name;
            this.description = data.description;
            
            this.order = data.order;

            this.type = new CustomPropertyType(data.type);
            this.customPropertyHasCustomFields =
                data.customPropertyHasCustomFields ?
                data.customPropertyHasCustomFields.map(_ => new CustomPropertyHasCustomFields(_)) :
                [];
        }
    }
}